src/content/structured/get-started/testing-with-jest-and-rtl.mdx

```diff
@@ -0,0 +1,178 @@
+    rel="noreferer noopener nofollow"
+  >
+    'Getting Started'
+  </ic-link>{" "}
+  page.
+</p>
+
+<p>
+  More information on setting up Jest can be found on{" "}
+  <ic-link
+    target="_blank"
+    href="https://jestjs.io/docs/getting-started"
+    rel="noreferer noopener nofollow"
+  >
+    Jest’s Getting Started page
+  </ic-link>
+  .
+</p>
+
+## Testing with React components
+
+The `@ukic/react` package will need to be transformed before you can use these components in Jest tests.
+
+Add a `transformIgnorePatterns` field with the value `["/node_modules/(?!@ukic/react)"]` to your Jest config.
+
+## About React Testing Library
+
+<p>
+  <ic-link
+    target="_blank"
+    href="https://testing-library.com/docs/react-testing-library/intro"
+    rel="noreferer noopener nofollow"
+  >
+    React Testing Library (RTL)
+  </ic-link>{" "}
+  provides a set of utility functions, which interrogate the DOM tree, in order
+  to build maintainable React tests. The recommended approach with React Testing
+  Library is to test React components based on how the users view the components
+  rather than the implementation of the component. For example, a user would
+  view a button by the label 'Add', so the library provides a method called{" "}
+  <code style={{ backgroundColor: "#EEEFF0" }}>getByText()</code> to facilitate
+  that.
+</p>
+
+## Shadow DOM Testing Library
+
+<p>
+  It is not possible to exclusively use RTL to test{" "}
+  <code style={{ backgroundColor: "#EEEFF0" }}>@ukic/react</code> components as
+  they are React-wrapped web components, which use the{" "}
+  <ic-link
+    target="_blank"
+    href="https://developer.mozilla.org/en-US/docs/Web/Web_Components/Using_shadow_DOM"
+    rel="noreferer noopener nofollow"
+  >
+    shadow DOM
+  </ic-link>{" "}
+  (<code style={{ backgroundColor: "#EEEFF0" }}>&lt;ShadowRoot&gt;</code>). RTL
+  does not provide utility functions that traverse beyond the shadow DOM. With
+  the addition of{" "}
+  <ic-link
+    target="_blank"
+    href="https://github.com/KonnorRogers/shadow-dom-testing-library"
+    rel="noreferer noopener nofollow"
+  >
+    Shadow DOM Testing Library
+  </ic-link>
+  , elements within the shadow DOM can be selected and then interacted with by using
+  methods provided by RTL. Shadow DOM Testing Library provides functions with 'Shadow'
+  prefixed to the query type (for example, <code
+    style={{ backgroundColor: "#EEEFF0" }}
+  >
+    getByShadowText()
+  </code>).
+</p>
+
+```tsx
+// DOM tree for IcAlert.
+<ic-alert class="dark hydrated" message="Foo" role="alert">
+  <ShadowRoot>
+    ...
+    <ic-typography class="ic-typography-body hydrated">
+      <ShadowRoot>
+        <slot />
+      </ShadowRoot>
+      Foo
+    </ic-typography>
+    ...
+  </ShadowRoot>
+</ic-alert>
+```
+
+## Examples with multiple components
+
+Below is an example of testing @ukic/react components within an ICDS ‘test app’ using RTL, Jest and Shadow DOM Testing Library.
```

Comment:
The formatting of @ukic/react here is different to other instances - think it's just that it's missing the back ticks.